src/main/scala/com/cognite/spark/datasource/EventsRelation.scala

```diff
@@ -180,52 +178,7 @@ class EventsRelation(config: RelationConfig)(@transient val sqlContext: SQLConte
     (postUpdate, postNewItems).parMapN((_, _) => ())
   }
 
-  case class ColumnFilter(
-      column: String
-  )
-
-  // scalastyle:off cyclomatic.complexity
-  def getFilter(filter: Filter, colName: String): Seq[ColumnFilter] =
-    filter match {
-      case IsNotNull(`colName`) => Seq()
-      case EqualTo(`colName`, value) => Seq(ColumnFilter(value.toString))
-      case EqualNullSafe(`colName`, value) => Seq(ColumnFilter(value.toString))
-      case In(`colName`, values) => values.map(v => ColumnFilter(v.toString))
-      case And(f1, f2) => getFilter(f1, colName) ++ getFilter(f2, colName)
-      case Or(f1, f2) => getFilter(f1, colName) ++ getFilter(f2, colName)
-      // TODO: add support for String filtering using the 'Search for events' endpoint
-      case StringStartsWith(`colName`, value) =>
-        sys.error(
-          s"Filtering using 'string starts with' not allowed for events, attempted for ${value.toString}")
-      case StringEndsWith(`colName`, value) =>
-        sys.error(
-          s"Filtering using 'string ends with' not allowed for events, attempted for ${value.toString}")
-      case StringContains(`colName`, value) =>
-        sys.error(
-          s"Filtering using 'string contains' not allowed for data points, attempted for ${value.toString}")
-      case _ =>
-        Seq()
-    }
-
-  // scalastyle:on cyclomatic.complexity
-  override def buildScan(): RDD[Row] = buildScan(Array.empty, Array.empty)
-
-  override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
-    val types = filters.flatMap(getFilter(_, "type")).map(_.column)
-    val subTypes = filters.flatMap(getFilter(_, "subtype")).map(_.column)
-
-    val eventFilters = for {
-      eventType <- if (types.isEmpty) { Array(None) } else { types.map(Some(_)) }
-      eventSubType <- if (subTypes.isEmpty) { Array(None) } else { subTypes.map(Some(_)) }
-    } yield EventFilter(eventType, eventSubType)
-
-    EventsRdd(sqlContext.sparkContext, listUrl(), (e: EventItem) => {
-      if (config.collectMetrics) {
-        eventsRead.inc()
-      }
-      toRow(e, requiredColumns)
-    }, eventFilters, config, cursors())
-  }
+  override val fieldsWithPushdownFilter: Seq[String] = Seq("source", "type", "assetIds")
```

Comment:
`assetIds` is new, we should have some tests for that.

Could we also add push-down filtering for the timestamp columns while we're at this? Updated, created, start time, end time. Would be very useful.

Also id, if that makes sense (may be more difficult?)

Reply:
Push-down filtering for timestamp should be fine, but will require some custom work for Events so let's put that in another PR.

I'm not sure about ID, for 0.6 we would have to use the /events/{eventId} endpoint. We should probably just let ID take precedence over other pushdown filters since there'll be just a few results anyways.

Reply:
Looks like `assetIds` will need special treatment as well, I'll remove it from here and add an issue for it.

Reply:
`subType` however should be easy, right?

Reply:
`subType` will throw an error unless the API call has a filter on `type` as well, which doesn't fit too easily with the current implementation (one API call per pushdown filter).